### PR TITLE
PYCO-15: Add API Documentation

### PR DESCRIPTION
**Pull request by thejcfactor**

### CODE REVIEW

File: docs/couchbase_columnar_api/errors.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+++++++++++++++++++++++++++++++++
+.. autoclass:: ColumnarError
+
+InvalidCredentialError
+++++++++++++++++++++++++++++++++
+.. autoclass:: InvalidCredentialError
+
+QueryError
+++++++++++++++++++++++++++++++++
+.. autoclass:: QueryError
```
Comment (DemetrisChr) [inline docs/couchbase_columnar_api/errors.rst:21]: It appears that the `server_message` and `code` properties are missing from the generated HTML docs

File: docs/couchbase_columnar_api/results.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+==============
+Results
+==============
+
+.. module:: couchbase_columnar.result
+
+QueryResult
```
Comment (DemetrisChr) [inline docs/couchbase_columnar_api/results.rst:7]: Do we need `QueryResult` in the docs, if it's not a class we expect users to use directly?

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+:doc:`couchbase_columnar_api/credential`
+   API reference for Credential.
+
+:doc:`couchbase_columnar_api/errors`
+   API reference for Errors.
+
+:doc:`couchbase_columnar_api/options`
+   API reference for operation options.
+
+:doc:`couchbase_columnar_api/results`
```
Comment (DemetrisChr) [inline docs/index.rst:36]: This makes `BlockingQueryResult` appear in the Global API, which is synchronous-API specific

File: docs/couchbase_columnar_api/errors.rst (hunk 1 above)
Comment (thejcfactor) [reply docs/couchbase_columnar_api/errors.rst:21]: Updated 👍

File: docs/index.rst (hunk 3 above)
Comment (thejcfactor) [reply docs/index.rst]: Updated 👍

File: docs/couchbase_columnar_api/results.rst (hunk 2 above)
Comment (thejcfactor) [reply docs/couchbase_columnar_api/results.rst]: Fair point, removed in latest updated 👍

**Review by DemetrisChr — APPROVED** [review]

**Pull request merged**